gammapy/visualization/utils.py

```diff
@@ -178,3 +181,135 @@ def plot_theta_squared_table(table):
     ax1.errorbar(x, table["sqrt_ts"], xerr=xerr, linestyle="None")
     ax1.set_xlabel(f"Theta [{theta2_axis.unit.to_string(UNIT_STRING_FORMAT)}]")
     ax1.set_ylabel("Significance")
+
+
+def plot_distribution(
+    wcs_map,
+    ax=None,
+    ncols=3,
+    fit=True,
+    func=None,
+    kwargs_hist=None,
+    kwargs_axes=None,
+    kwargs_fit=None,
+):
+    """
+    Plot the 1D distribution of data inside a map as an histogram. If the dimension of the map is smaller than 2,
+    a unique plot will be displayed. Otherwise, if the dimension is 3 or greater, a grid of plot will be displayed.
+
+    Parameters
+    ----------
+    wcs_map : an instance of `~gammapy.maps.WcsNDMap`
+        A map that contains data to be plotted.
+    ax : `~matplotlib.axes.Axes` or list of `~matplotlib.axes.Axes`
+        Axis object to plot on. If a list of Axis is provided it has to be the same length as the length of _map.data.
+    ncols : int
+        Number of columns to plot if a "plot grid" was to be done.
+    fit : bool
+        Whether to perform a fit of the distribution of data. If True, the fit is performed by `scipy.stats`.
+    func : function object
+        The function to pass to `scipy.optimize.curve_fit`
```

Comment:
I am using `curve_fit` because you can have access to the covariance matrix of the fit, therefore to the error of the parameters. Moreover, I think it is more flexible, user can input some fancy function that they would have define before. The fit will basically work with every function.